packages/core/src/Subscribe.tsx

```diff
@@ -33,11 +33,15 @@ export const Subscribe: React.FC<{
         throw e
       }),
     )
-    setMounted(1)
+    setMounted(source$)
```

Comment:
Wouldn't it be better to set the state in a conditional clause on render? :thinking: 

Something, like this?

```tsx
const init = (source$: Observable<any>): [boolean, Observable<any>] => {
  const result = [true, source$] as const
  try {
    ;(source$ as any).gV()
  } catch (e) {
    if (!e.then) result[0] = false
  }
  return result
}

export const Subscribe: React.FC<{
  source$: Observable<any>
  fallback?: NonNullable<ReactNode> | null
}> = ({ source$, children, fallback }) => {
  const [state, setState] = useState(() => init(source$))

  if (state[1] !== source$) setState(init(source$))

  useLayoutEffect(() => {
    const subscription = source$.subscribe(noop, (e) =>
      setState(() => {
        throw e
      }),
    )
    setState(current => current[0] ? current : [true, current[1]])
    return () => {
      subscription.unsubscribe()
    }
  }, [source$])

  return (
    <Suspense fallback={fallback || null}>
      {state[0] ? children : <Throw />}
    </Suspense>
  )
}
```